commit 3c1a70ea1e0c
Author: Paolo Amadini <paolo.mozmail@amadzone.org>
Date:   Tue Jan 30 14:38:25 2018 +0000

    Bug 1424469 - The Downloads view in the Library window doesn't get focus when opened. r=mak, a=RyanVM
    
    We have to keep track of the focused element while we remove the richlistbox element temporarily.
    
    MozReview-Commit-ID: AePydUyyUS7
    
    --HG--
    extra : source : 6aa2d7b4215e350c76bdfbb9132eb5ae42b23123
---
 browser/components/downloads/content/allDownloadsViewOverlay.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git browser/components/downloads/content/allDownloadsViewOverlay.js browser/components/downloads/content/allDownloadsViewOverlay.js
index 64052c0259f1..7f1990651ced 100644
--- browser/components/downloads/content/allDownloadsViewOverlay.js
+++ browser/components/downloads/content/allDownloadsViewOverlay.js
@@ -1022,11 +1022,15 @@ DownloadsPlacesView.prototype = {
       xblFields.set(key, value);
     }
 
+    let oldActiveElement = document.activeElement;
     let parentNode = this._richlistbox.parentNode;
     let nextSibling = this._richlistbox.nextSibling;
     parentNode.removeChild(this._richlistbox);
     this._richlistbox.appendChild(aDOMFragment);
     parentNode.insertBefore(this._richlistbox, nextSibling);
+    if (oldActiveElement && oldActiveElement != document.activeElement) {
+      oldActiveElement.focus();
+    }
 
     for (let [key, value] of xblFields) {
       this._richlistbox[key] = value;
